Lib/tkinter/test/test_tkinter/test_misc.py

```diff
@@ -48,6 +48,47 @@ def test_tk_setPalette(self):
                 '^must specify a background color$',
                 root.tk_setPalette, highlightColor='blue')
 
+    def test_after_info(self):
+        root = self.root
+
+        # No events.
+        self.assertEqual(root.after_info(), ())
+
+        # Add timer.
+        timer = root.after(1, lambda: 'break')
+
+        # With no parameter, it returns a tuple of the event handler ids.
+        self.assertEqual(root.after_info(), (timer, ))
+        # Process event to remove it.
+        root.update()
+
+        timer1 = root.after(5000, lambda: 'break')
+        timer2 = root.after(5000, lambda: 'break')
+        idle1 = root.after_idle(lambda: 'break')
+        # Only contains new events and not 'timer'.
+        self.assertEqual(root.after_info(), (idle1, timer2, timer1))
```

Comment:
Is this order for the return tuple guaranteed, or could it fail on some system or tk version?

Reply:
I don't know.

Reply:
Maybe the Tcl code might help answer that?  Does it traverse a linked list?
```
	    for (afterPtr = assocPtr->firstAfterPtr; afterPtr != NULL;
		    afterPtr = afterPtr->nextPtr) {
		if (assocPtr->interp == interp) {
                    Tcl_ListObjAppendElement(NULL, resultObj, Tcl_ObjPrintf(
                            "after#%d", afterPtr->id));
		}
	    }
            Tcl_SetObjResult(interp, resultObj);
	    return TCL_OK;
```
And this is how it's built:
```
	afterPtr->nextPtr = assocPtr->firstAfterPtr;
	assocPtr->firstAfterPtr = afterPtr;
```